pkg/sdk/stages_def.go

```diff
@@ -0,0 +1,401 @@
+)
+
+type ExternalStageS3EncryptionOption string
+
+var (
+	ExternalStageS3EncryptionCSE    ExternalStageS3EncryptionOption = "AWS_CSE"
+	ExternalStageS3EncryptionSSES3  ExternalStageS3EncryptionOption = "AWS_SSE_S3"
+	ExternalStageS3EncryptionSSEKMS ExternalStageS3EncryptionOption = "AWS_SSE_KMS"
+	ExternalStageS3EncryptionNone   ExternalStageS3EncryptionOption = "NONE"
+)
+
+type ExternalStageGCSEncryptionOption string
+
+var (
+	ExternalStageGCSEncryptionSSEKMS ExternalStageGCSEncryptionOption = "GCS_SSE_KMS"
+	ExternalStageGCSEncryptionNone   ExternalStageGCSEncryptionOption = "NONE"
+)
+
+type ExternalStageAzureEncryptionOption string
+
+var (
+	ExternalStageAzureEncryptionCSE  ExternalStageAzureEncryptionOption = "AZURE_CSE"
+	ExternalStageAzureEncryptionNone ExternalStageAzureEncryptionOption = "NONE"
+)
+
+type StageCopyColumnMapOption string
+
+var (
+	StageCopyColumnMapCaseSensitive   StageCopyColumnMapOption = "CASE_SENSITIVE"
+	StageCopyColumnMapCaseInsensitive StageCopyColumnMapOption = "CASE_INSENSITIVE"
+	StageCopyColumnMapCaseNone        StageCopyColumnMapOption = "NONE"
+)
+
+func createStageOperation(structName string, apply func(qs *g.QueryStruct) *g.QueryStruct) *g.QueryStruct {
+	qs := g.NewQueryStruct(structName).
+		Create().
+		OrReplace().
+		OptionalSQL("TEMPORARY").
+		SQL("STAGE").
+		IfNotExists().
+		Name()
+	qs = apply(qs)
+	return qs.
+		OptionalQueryStructField("FileFormat", stageFileFormatDef, g.ListOptions().Parentheses().SQL("FILE_FORMAT =")).
+		OptionalQueryStructField("CopyOptions", stageCopyOptionsDef, g.ListOptions().Parentheses().SQL("COPY_OPTIONS =")).
+		OptionalComment().
+		OptionalTags().
+		WithValidation(g.ConflictingFields, "OrReplace", "IfNotExists")
+}
+
+func alterStageOperation(structName string, apply func(qs *g.QueryStruct) *g.QueryStruct) *g.QueryStruct {
+	qs := g.NewQueryStruct(structName).
+		Alter().
+		SQL("STAGE").
+		IfExists().
+		Name().
+		SQL("SET")
+	qs = apply(qs)
+	return qs.
+		OptionalQueryStructField("FileFormat", stageFileFormatDef, g.ListOptions().Parentheses().SQL("FILE_FORMAT =")).
+		OptionalQueryStructField("CopyOptions", stageCopyOptionsDef, g.ListOptions().Parentheses().SQL("COPY_OPTIONS =")).
+		OptionalComment().
+		WithValidation(g.ValidIdentifier, "name")
+}
+
+var stageFileFormatDef = g.NewQueryStruct("StageFileFormat").
+	OptionalTextAssignment("FORMAT_NAME", g.ParameterOptions().SingleQuotes()).
+	OptionalAssignment("TYPE", g.KindOfTPointer[FileFormatType](), g.ParameterOptions()).
+	PredefinedQueryStructField("Options", g.KindOfTPointer[FileFormatTypeOptions](), g.ListOptions().NoComma())
+
+var stageCopyOptionsDef = g.NewQueryStruct("StageCopyOptions").
+	OptionalQueryStructField(
+		"OnError",
+		g.NewQueryStruct("StageCopyOnErrorOptions").
+			OptionalSQL("CONTINUE").
+			OptionalSQL("SKIP_FILE").
+			//OptionalSQL("SKIP_FILE_n").  // TODO templated value - not even supported by structToSQL
```

Comment:
We have a few options with this one
- enhance sql_builder to be able to make it work
- leave it as is with a comment e.g. in the readme that this should be eventually fixed
- replace the whole struct with string statement

Reply:
For me the second option for now.